lib/elastic_apm/metrics/metric.rb

```diff
@@ -57,6 +57,8 @@ def collect
         @mutex.synchronize do
           collected = @value
 
+          return nil if collected.is_a?(Float) && collected.nan?
```

Comment:
How about checking `collected.finite?` just to cover all bases?
https://ruby-doc.org/core-2.7.0/Float.html#method-i-finite-3F

Reply:
```suggestion
          return nil if collected.is_a?(Float) && !collected.finite?
```

Reply:
Good suggestion!